lib/seahorn/BvOpSem2.cc

```diff
@@ -2229,10 +2233,31 @@ APInt Bv2OpSem::agg(Type *aggTy, const std::vector<GenericValue> &elements,
   const StructLayout *SL = getDataLayout().getStructLayout(STy);
   for (int i = 0 ; i <  elements.size(); i++) {
     const GenericValue element = elements[i];
+    Type *ElmTy = STy->getElementType(i);
     if (element.AggregateVal.empty()) {
-      next = element.IntVal;
-    } else { // assuming only dealing with int as terminal struct element
-      next = agg(STy->getElementType(i), element.AggregateVal, ctx);
+      // Assuming only dealing with Int or Pointer as struct terminal elements
+      if (ElmTy->isIntegerTy())
+        next = element.IntVal;
+      else if (ElmTy->isPointerTy()){
+        auto ptrBv = reinterpret_cast<intptr_t>(GVTOP(element));
+        next = APInt(getDataLayout().getTypeSizeInBits(ElmTy), ptrBv);
+      } else {
+        // this should be handled in constant evaluation step
+        LOG("opsem",
+            WARN << "Unsupported type " << *ElmTy << " to convert in aggregate. \n";);
```

Comment:
`Unsupported` should be lower case. No `\n` at the end, added automatically by `WARN`